### ISSUE TITLE
Feature request: Allow quokka to profile async code.

### ISSUE THREAD

**Issue by bugs181** [wallabyjs/quokka#315]
### Issue description or question
As the title implies, I would like to profile async/callback code. This should be possible as seen in other tools of this nature.

### Sample code

```javascript
function someFn(cb) {
  setTimeout(function() { // ?.
    cb()
  }, 20 * 1000)
}

someFn(function() { // ?.

}) // ?.
```

### Example output

![quokka](https://user-images.githubusercontent.com/3699516/50956301-f2991980-1480-11e9-97c2-42f35da4d098.png)

### Quokka.js Console Output

```
​​​​Quokka #1 (node: v11.0.0)​​​​

​​​​​0.719ms​​​​​ at ​​​someFn(function () { })​​​ ​quokka.js:7:0​
```

### Code editor version

Visual Studio Code `v1.30.1`

### OS name and version

OSX Mojave `v10.14`

**Comment by ArtemGovorov** [comment]
A quick note - async/await code profiling is actually supported:

```js
(async () => {
    await (new Promise(resolve => setTimeout(() => resolve({ }), 300)))//?.
})()
```

<img width="718" alt="screen shot 2019-01-10 at 6 49 40 pm" src="https://user-images.githubusercontent.com/979966/50956860-83670800-1508-11e9-948f-1169ed615bb0.png">

Callbacks are tricky though, because we would need some more context from user to correctly understand when user wants to start/stop the timer. 

For example, you may profile the code with callbacks with `console.time`:

<img width="304" alt="screen shot 2019-01-10 at 7 07 17 pm" src="https://user-images.githubusercontent.com/979966/50957892-1ef97800-150b-11e9-8516-c78119f20545.png">

We could introduce a similar construct to start/stop the timer (and display the execution time inline). For example: `//?.s` to start the timer, and `//?.e` to stop it and print the time. Would something like this work for you?

> This should be possible as seen in other tools of this nature.

Could you please share what tools are you referring to?

**Comment by bugs181** [comment]
Absolutely! That's a compromise I'm willing to work with. I believe your other product Wallaby combined will Mocha will give you runtime testing/profiling for code. I abuse a Mocha test in this way to provide that information.

![wallaby](https://user-images.githubusercontent.com/3699516/50958468-638f0000-1486-11e9-9910-df84c20bc4db.png)
![mocha](https://user-images.githubusercontent.com/3699516/50958480-6a1d7780-1486-11e9-8d4c-4ddfef3ece50.png)

**Comment by bugs181** [comment]
Bump.
